pydatalab/src/pydatalab/bokeh_plots.py

```diff
@@ -62,6 +62,36 @@
   link.click();
   document.body.removeChild(link);
 """
+GENERATE_XY_CALLBACK = """
+  let columns = Object.keys(source.data);
+  console.log(columns);
+
+  let x_column = '2θ (°)';
+  let y_column = 'intensity';
+
+  if (!columns.includes(x_column)) {
+    x_column = columns[0];
+  }
+  if (!columns.includes(y_column)) {
+    y_column = columns[columns.includes('intensity') ? columns.indexOf('intensity') : 1];
+  }
+
+  const x_values = source.data[x_column];
+  const y_values = source.data[y_column];
+
+  var xyContent = "data:text/plain;charset=utf-8,";
+  for (var i = 0; i < x_values.length; i++) {
+    xyContent += x_values[i].toFixed(5) + "\\t" + y_values[i].toFixed(3) + "\\n";
+  }
+
+  const encodedUri = encodeURI(xyContent);
+  const link = document.createElement("a");
+  link.setAttribute("href", encodedUri);
+  link.setAttribute("download", "data.xy");
+  document.body.appendChild(link);
+  link.click();
+  document.body.removeChild(link);
+"""
```

Comment:
Can we figure out if we can add some of the sample info into this header, as per Matt's comment? https://github.com/datalab-org/datalab/issues/1272#issuecomment-3053848859

I'm not sure if this is currently possible, but I think the block at least knows the item ID, so can just say `"'Generated by datalab from file xyz.xrdml, wavelength lambda for sample <ID>"` as the first line and see if it works in Topas

Reply:
Done 👍